content/blog/devmode-vs-prodmode.md

```diff
@@ -0,0 +1,31 @@
+---
+date: 2024-06-30
+title: Production Mode vs. Dev Mode
+---
+
+When setting up workflow orchestration, it's crucial to understand the differences between production mode and development (dev) mode, particularly in terms of infrastructure requirements. This distinction ensures that workflows are efficiently managed and executed based on their intended use case. Here, we'll explore these differences, focusing on the infrastructure required to run workflows in each mode.
+
+# Production Mode
+Production mode is tailored for environments where stability, reliability, and scalability are paramount. The Orchestrator Helm chart or Orchestrator operator is designed specifically to meet the demanding requirements of production environments. Key requirements include:
+
+* **Long-running Workflows**: Production mode supports workflows that may take several hours or even days to complete, ensuring that these processes run smoothly without interruption.
+* **Persistence of Running Workflow Instances**: Ensuring the persistence of workflow instances is critical. In production, all workflow data is stored and maintained even if the orchestrator restarts, preventing data loss.
+* **Event Handling Reliability**: Reliable handling of events is essential for maintaining workflow integrity and ensuring that all triggers and actions occur as expected.
+* **Scalability**: The system must be capable of scaling up to handle increasing workloads, allowing for the addition of resources as demand grows.
+* **Updates Through Pipelines**: Workflows can be updated and managed through continuous integration/continuous deployment (CI/CD) pipelines, facilitating smooth and efficient updates.
+* **Externalizing Credentials**: Credentials are managed outside the workflow configuration to enhance security and simplify updates.
+* **Runtime Isolation**: Each workflow runs in its isolated environment, preventing any interference between workflows.
+* **Authorization and Administration**: Robust mechanisms are in place for authorizing and administering workflow deployments, ensuring that only authorized personnel can make changes.
```

Comment:
how does the authorization work? you mean the same as any k8s CR?

Reply:
yes, this is related to the workflow deployment (k8s RBAC to manage the sonataflow CRs)